### PR TITLE
Add the Convertible interface.

### PR DESCRIPTION
**Pull request by mattmoor**
This sketches the "Convertible" interface that @vagababov outlined in
last week's Serving API WG meeting.

See also: https://docs.google.com/document/d/1UFUfwIF4q4r6JVigom0jQivbcI8mFycWPc9z48yjSIc/edit#heading=h.9np2by1b65b

/lint

### CODE REVIEW

File: apis/interfaces.go [hunk 1]
```diff
@@ -35,6 +35,16 @@ type Validatable interface {
 	Validate(context.Context) *FieldError
 }
 
+// Convertible indicates that a particular type supports conversions to/from
+// "higher" versions of the same type.
+type Convertible interface {
```
Comment (vagababov) [inline apis/interfaces.go]: Do we care to specify whether we're doing staged a⇒b⇒c or a⇒c, or note that how a is converted to c is not prescribed and is an implementation detail?

Comment (mattmoor) [reply apis/interfaces.go]: I think that beyond the direction it's an implementation detail for you get from `a` to `c`.

Comment (vagababov) [reply apis/interfaces.go]: Well, then it would make sense to say, that any version above/below `x` is acceptable

**Comment by vagababov** [comment]
/lgtm
/hold
just in case anyone else has comments.

**Pull request merged**